src/sentry/auth/access.py

```diff
@@ -47,68 +54,141 @@ def get_permissions_for_user(user_id: int) -> FrozenSet[str]:
+        """
+        return scope in self.scopes
+
+    def get_organization_role(self) -> OrganizationRole | None:
+        if self.role is not None:
+            return cast(OrganizationRole, organization_roles.get(self.role))
+        return None
+
+    @abc.abstractmethod
+    def has_team_access(self, team: Team) -> bool:
+        """
+        Return bool representing if a user should have access to information for the given team.
+        >>> access.has_team_access(team)
+        """
+        pass
+
+    @abc.abstractmethod
+    def has_team_scope(self, team: Team, scope: str) -> bool:
+        pass
+
+    def has_team_membership(self, team: Team) -> bool:
+        return team.id in self.team_ids_with_membership
+
+    @abc.abstractmethod
+    def get_team_role(self, team: Team) -> TeamRole | None:
+        pass
+
+    @abc.abstractmethod
+    def has_project_access(self, project: Project) -> bool:
+        """
+        Return bool representing if a user should have access to information for the given project.
+        >>> access.has_project_access(project)
+        """
+        raise NotImplementedError
+
+    def has_projects_access(self, projects: Iterable[Project]) -> bool:
+        """
+        Returns bool representing if a user should have access to every requested project
+        """
+        return all([self.has_project_access(project) for project in projects])
+
+    def has_project_membership(self, project: Project) -> bool:
+        """
+        Return bool representing if a user has explicit membership for the given project.
+        >>> access.has_project_membership(project)
+        """
+        return project.id in self.project_ids_with_team_membership
+
+    def has_project_scope(self, project: Project, scope: str) -> bool:
+        """
+        Return bool representing if a user should have access with the given scope to information
+        for the given project.
 
-    sso_is_valid = auth_identity.is_valid(member)
-    return SsoState(True, sso_is_valid)
+        >>> access.has_project_scope(project, 'project:read')
+        """
+        return self.has_any_project_scope(project, [scope])
+
+    @abc.abstractmethod
+    def has_any_project_scope(self, project: Project, scopes: Collection[str]) -> bool:
+        pass
+
+    def to_django_context(self) -> Mapping[str, bool]:
+        return {s.replace(":", "_"): self.has_scope(s) for s in settings.SENTRY_SCOPES}
 
 
 @dataclass
-class Access(abc.ABC):
+class DbAccess(Access, abc.ABC):
     # TODO(dcramer): this is still a little gross, and ideally backend access
     # would be based on the same scopes as API access so there's clarity in
     # what things mean
 
-    _sso_state: SsoState = SsoState(False, False)
-
-    @property
-    def sso_is_valid(self) -> bool:
-        return self._sso_state.is_valid
-
-    @property
-    def requires_sso(self) -> bool:
-        return self._sso_state.is_required
+    sso_is_valid: bool = False
```

Comment:
A little bit of a revert due to conflicts from @RyanSkonnord 's merge, and my own work.  I did end up using SsoState, but I baked into the service side implementation and left this implementation "as is".